pkg/kubectl/cmd/edit.go

```diff
@@ -145,139 +145,142 @@ func RunEdit(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []strin
 	defaultVersion := cmdutil.OutputVersion(cmd, clientConfig.Version)
 	results := editResults{}
 	for {
-		obj, err := resource.AsVersionedObject(infos, false, defaultVersion)
+		objs, err := resource.AsVersionedObjects(infos, defaultVersion)
 		if err != nil {
 			return preservedFile(err, results.file, out)
 		}
+		for _, obj := range objs {
+			// TODO: add an annotating YAML printer that can print inline comments on each field,
+			//   including descriptions or validation errors
+
+			// generate the file to edit
+			buf := &bytes.Buffer{}
+			if err := results.header.writeTo(buf); err != nil {
+				return preservedFile(err, results.file, out)
+			}
+			if err := printer.PrintObj(obj, buf); err != nil {
+				return preservedFile(err, results.file, out)
+			}
+			original := buf.Bytes()
 
-		// TODO: add an annotating YAML printer that can print inline comments on each field,
-		//   including descriptions or validation errors
-
-		// generate the file to edit
-		buf := &bytes.Buffer{}
-		if err := results.header.writeTo(buf); err != nil {
-			return preservedFile(err, results.file, out)
-		}
-		if err := printer.PrintObj(obj, buf); err != nil {
-			return preservedFile(err, results.file, out)
-		}
-		original := buf.Bytes()
-
-		// launch the editor
-		edit := editor.NewDefaultEditor()
-		edited, file, err := edit.LaunchTempFile("kubectl-edit-", ext, buf)
-		if err != nil {
-			return preservedFile(err, results.file, out)
-		}
+			// launch the editor
+			edit := editor.NewDefaultEditor()
+			edited, file, err := edit.LaunchTempFile("kubectl-edit-", ext, buf)
+			if err != nil {
+				return preservedFile(err, results.file, out)
+			}
 
-		// cleanup any file from the previous pass
-		if len(results.file) > 0 {
-			os.Remove(results.file)
-		}
+			// cleanup any file from the previous pass
+			if len(results.file) > 0 {
+				os.Remove(results.file)
+			}
 
-		glog.V(4).Infof("User edited:\n%s", string(edited))
-		fmt.Printf("User edited:\n%s", string(edited))
-		lines, err := hasLines(bytes.NewBuffer(edited))
-		if err != nil {
-			return preservedFile(err, file, out)
-		}
-		if bytes.Equal(original, edited) {
-			if len(results.edit) > 0 {
-				preservedFile(nil, file, out)
-			} else {
-				os.Remove(file)
+			glog.V(4).Infof("User edited:\n%s", string(edited))
+			fmt.Printf("User edited:\n%s", string(edited))
+			lines, err := hasLines(bytes.NewBuffer(edited))
+			if err != nil {
+				return preservedFile(err, file, out)
 			}
-			fmt.Fprintln(out, "Edit cancelled, no changes made.")
-			return nil
-		}
-		if !lines {
-			if len(results.edit) > 0 {
-				preservedFile(nil, file, out)
-			} else {
-				os.Remove(file)
+			if bytes.Equal(original, edited) {
```

Comment:
Note that this compares comments too. We need something like https://github.com/kargakis/origin/blob/e65b2db16414459318a03e07de0fcd6540e5a655/pkg/cmd/cli/cmd/edit.go#L215